client/src/store/auth/operations.ts

```diff
@@ -1,30 +1,45 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import axios from 'axios';
+import axios, { AxiosError } from 'axios';
 
 import { RootState } from '@store';
+import { store } from '@store';
 import { LoggedUser } from '@types';
 
-axios.defaults.baseURL = 'https://todo-app-backend-igep.onrender.com/api';
+import { logout } from './slice';
 
-const setAuthHeader = (token: string) => {
-  axios.defaults.headers.common.Authorization = `Bearer ${token}`;
-};
+export const instance = axios.create({
+  baseURL: 'https://todo-app-backend-igep.onrender.com/api',
+});
 
-export const clearAuthHeader = () => {
-  axios.defaults.headers.common.Authorization = '';
+export const setAuthHeader = (token?: string) => {
+  if (token) {
+    instance.defaults.headers.common.Authorization = `Bearer ${token}`;
+  } else {
+    delete instance.defaults.headers.common.Authorization;
+  }
 };
 
+instance.interceptors.response.use(
+  response => response,
+  (error: AxiosError) => {
+    if (error.response && error.response.status === 401) {
+      store.dispatch(logout());
+    }
+    return Promise.reject(error);
+  }
+);
```

Comment:
Could be better to have this in some general file as it's related not only to auth